File: app/src/main/java/com/binkery/itarget/sqlite/ItemEntity.java
package com.binkery.itarget.sqlite;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * Create by binkery@gmail.com
 * on 2019 08 08
 * Copyright (c) 2019 iTarget.binkery.com. All rights reserved.
 */
@Entity(tableName = "items")
public class ItemEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    public int id;

    @ColumnInfo(name = "uuid")
    public String uuid;

    @ColumnInfo(name = "target_id")
    public int targetId;

    @ColumnInfo(name = "start_time")
    public long startTime;

    @ColumnInfo(name = "end_time")
    public long endTime;

    @ColumnInfo(name = "value")
    public long value;

    @ColumnInfo(name = "content")
    public String content;
}

File: ipassword/src/main/java/com/binkery/ipassword/sqlite/ItemEntity.kt
package com.binkery.ipassword.sqlite

import android.content.Context
import androidx.room.*

@Entity(tableName = "items")
class ItemEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    var id: Int = 0

    @ColumnInfo(name = "name")
    var name: String = ""

    @ColumnInfo(name = "username")
    var username: String = ""

    @ColumnInfo(name = "password")
    var password: String = ""

    @ColumnInfo(name = "comments")
    var comments: String = ""

}

@Dao
interface ItemDao {

    @Insert
    fun insert(item: ItemEntity)

    @Update
    fun update(item: ItemEntity)

    @Delete
    fun delete(item: ItemEntity)

    @Query("select * from items")
    fun queryAll(): List<ItemEntity>

    @Query("select * from items where _id = :id")
    fun queryById(id: Int): ItemEntity


}

@Database(entities = arrayOf(ItemEntity::class), version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun itemDao(): ItemDao
}

class DBHelper private constructor() {

    companion object {
        val instance: DBHelper = Holder.instance
    }

    private object Holder {
        val instance = DBHelper()
    }

    private var mAppDatabase: AppDatabase? = null

    fun init(context: Context) {
        mAppDatabase = Room.databaseBuilder(context, AppDatabase::class.java, "ipassword")
                .allowMainThreadQueries()
                .build()
    }

    fun itemDao(): ItemDao {
        return mAppDatabase?.itemDao()!!
    }
}